festim/hydrogen_transport_problem.py

```diff
@@ -123,11 +129,50 @@ def initialise(self):
         self.t = fem.Constant(self.mesh.mesh, 0.0)
         self.dt = self.settings.stepsize.get_dt(self.mesh.mesh)
 
+        self.define_temperature()
         self.define_boundary_conditions()
         self.create_formulation()
         self.create_solver()
         self.defing_export_writers()
 
+    def define_temperature(self):
+        """If temperature value is given only dependent on t, create a
+        fem.Constant else create an expression to be updated and a function"""
```

Comment:
This method does a lot of things. I would be a bit more descriptive here. Something along these lines. What do you think?

```suggestion
        """Sets the value of .temperature_fenics_value.
        The type depends on self.temperature.
        If self.temperature is a function on t only, create a
        fem.Constant. Else create an dolfinx.fem.Expression
        (stored in self.temperature_expr) to be updated, a
        dolfinx.fem.Function object is created from the Expression
        (stored in self.temperature_fenics_value)"""
```